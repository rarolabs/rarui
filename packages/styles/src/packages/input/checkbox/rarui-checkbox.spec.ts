import { checkbox } from ".";

describe("Checkbox Styles", () => {
  it("should apply base styles correctly", () => {
    const className = checkbox.classnames.checkbox({ size: "medium" });
    expect(className).toContain("checkbox");
  });

  it("should apply size variants correctly", () => {
    const sizes = ["medium", "large"] as const;
    sizes.forEach((size) => {
      const className = checkbox.classnames.checkbox({ size });
      expect(className).toContain(size);
      expect(className).toMatchSnapshot(`size-${size}`);
    });
  });

  it("should apply error variant styles correctly", () => {
    const className = checkbox.classnames.checkbox({ error: true });
    expect(className).toContain("error");
    expect(className).toMatchSnapshot("error");
  });

  it("should apply indeterminate state styles correctly", () => {
    const className = checkbox.classnames.checkbox({ indeterminate: true });
    expect(className).toContain("indeterminate");
    expect(className).toMatchSnapshot("indeterminate");
  });

  it("should apply pseudo-class styles correctly", () => {
    const className = checkbox.classnames.checkbox({ size: "medium" });
    expect(className).toMatchSnapshot("pseudo-classes");
  });

  it("should apply label styles correctly", () => {
    const className = checkbox.classnames.label({ error: false });
    expect(className).toContain("label");
    expect(className).toMatchSnapshot("label");
  });

  it("should apply overlay styles correctly on hover", () => {
    const className = checkbox.classnames.overlay;
    expect(className).toMatchSnapshot("overlay");
  });

  it("should apply container styles correctly", () => {
    const className = checkbox.classnames.container;
    expect(className).toMatchSnapshot("container");
  });
});
