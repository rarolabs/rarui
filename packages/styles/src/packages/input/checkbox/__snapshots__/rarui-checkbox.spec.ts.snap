// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Styles should apply container styles correctly: container 1`] = `"rarui-checkbox_container__1qvfiyr8"`;

exports[`Checkbox Styles should apply error variant styles correctly: error 1`] = `"rarui-checkbox__1qvfiyr1 rarui-checkbox_error_true__1qvfiyr2"`;

exports[`Checkbox Styles should apply indeterminate state styles correctly: indeterminate 1`] = `"rarui-checkbox__1qvfiyr1 rarui-checkbox_indeterminate_true__1qvfiyr5"`;

exports[`Checkbox Styles should apply label styles correctly: label 1`] = `"rarui-checkbox_label__1qvfiyr6"`;

exports[`Checkbox Styles should apply overlay styles correctly on hover: overlay 1`] = `"rarui-checkbox_overlay__1qvfiyr9"`;

exports[`Checkbox Styles should apply pseudo-class styles correctly: pseudo-classes 1`] = `"rarui-checkbox__1qvfiyr1 rarui-checkbox_size_medium__1qvfiyr3"`;

exports[`Checkbox Styles should apply size variants correctly: size-large 1`] = `"rarui-checkbox__1qvfiyr1 rarui-checkbox_size_large__1qvfiyr4"`;

exports[`Checkbox Styles should apply size variants correctly: size-medium 1`] = `"rarui-checkbox__1qvfiyr1 rarui-checkbox_size_medium__1qvfiyr3"`;
